File: DoAn_Project1/Model/HETHONG/TAIKHOAN/Dtos/MODELTaiKhoanPhanQuyen.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Model.BASE;

namespace Model.HETHONG.TAIKHOAN.Dtos
{
    public class MODELTaiKhoanPhanQuyen 
    {
        public MODELTaiKhoan TaiKhoan { get; set; }
        public List<MODELMenuLogin> Menu { get; set; }
        public List<MODELPhanQuyen> PhanQuyen { get; set; }
        public List<MODELNhomQuyenLogin> NhomQuyen { get; set; }
    }

    public class MODELMenuLogin
    {
        public string ControllerName { get; set; }
        public string Action { get; set; }
        public string TenGoi { get; set; }
        public Guid NhomQuyenId { get; set; }
        public int? Sort { get; set; }
    }

    public class MODELNhomQuyenLogin
    {
        public Guid Id { get; set; }
        public string TenGoi { get; set; }

        public int? Sort { get; set; }
        public string Icon { get; set; }
    }
}

File: FrontEnd/src/app/MODEL/HETHONG/TAIKHOAN/modeltai-khoan-phan-quyen.ts
import { MODELPhanQuyen } from "./modelphan-quyen";
import { MODELTaiKhoan } from "./modeltai-khoan";

export class MODELTaiKhoanPhanQuyen {
    taiKhoan: MODELTaiKhoan | null = new MODELTaiKhoan;
    menu: MODELMenuLogin[] = [];
    phanQuyen: MODELPhanQuyen[] = [];
    nhomQuyen: MODELNhomQuyenLogin[] = [];
}
export class MODELMenuLogin {
    ControllerName!: string | null;
    Action!: string | null;
    TenGoi!: string | null;
    NhomQuyenId!: string | null;
    Sort!: number | null;
}
export class MODELNhomQuyenLogin{
    Id!: string | null;
    TenGoi!: string | null;
    Sort!: number | null;
    Icon!: string | null;
}
